### PR TITLE
Fixed some issues for RemoveZeroAndOneBinaryRector

### PR DESCRIPTION
**Pull request by jeroensmit**
Fixed issue where '1 / x' is considered to be equal to 'x'
Fixed issue where '($var + 1)' inside another expression is considered to be equal to '1'
Expanded unit test for ValueResolver

### CODE REVIEW

File: packages/DeadCode/src/Rector/Plus/RemoveZeroAndOneBinaryRector.php [hunk 1]
```diff
@@ -105,9 +105,11 @@ private function processBinaryPlusAndMinus(BinaryOp $binaryOp): ?Expr
      */
     private function processBinaryMulAndDiv(BinaryOp $binaryOp): ?Expr
     {
-        if ($this->isValue($binaryOp->left, 1)) {
-            if ($this->isNumberType($binaryOp->right)) {
-                return $binaryOp->right;
+        if ($binaryOp instanceof Mul) {
```
Comment (TomasVotruba) [inline packages/DeadCode/src/Rector/Plus/RemoveZeroAndOneBinaryRector.php]: This will skip all the `Div`, right?

Or will it change:

```diff
-5/1
+5
```

Comment (jeroensmit) [reply packages/DeadCode/src/Rector/Plus/RemoveZeroAndOneBinaryRector.php]: it will only skip the div for 1/5:
$binaryOp->left == 1 
$binaryOp->right == 5

Comment (TomasVotruba) [reply packages/DeadCode/src/Rector/Plus/RemoveZeroAndOneBinaryRector.php]: Could you add test case for it, just to be sure?

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**